_posts/2021-02-26-feather-nrf52840.md

```diff
@@ -0,0 +1,462 @@
+configuration.
+For simplicity I cut out the “Adafruit” out of the board's identifying name—and
+because the [Adafruit Feather M0] already was called `feather-m0`. Hence, for
+RIOT's purpuses the board is now just called `feather-nrf52840`. Under this name
+I created a directory for the board configuration to reside in:
+[`boards/feather-nrf52840`][directory].
+
+There I first created a Makefile to tell the build system [`make`][make] that
+the directory is indeed describing a board. This file is called
+[`boards/feather-nrf52840/Makefile`][Makefile]:
+
+{% highlight Makefile %}
+MODULE = board
+
+include $(RIOTBASE)/Makefile.base
+{% endhighlight %}
+
+The first line tells the build system, that this is a module called `board` (all
+boards are called like this) and the third includes all the stuff the build
+system developers of RIOT already provided for a module to be built.
+
+However, the build system needs more information. What features (i.e. interfaces
+and devices) does the board support? This is done with the file
+[`boards/feather-nrf52840/Makefile.features`][Makefile.features]:
+
+{% highlight Makefile %}
+CPU_MODEL = nrf52840xxaa
+
+# Put defined MCU peripherals here (in alphabetical order)
+FEATURES_PROVIDED += periph_i2c
+FEATURES_PROVIDED += periph_spi
+FEATURES_PROVIDED += periph_uart
+FEATURES_PROVIDED += periph_usbdev
+
+# Various other features (if any)
+FEATURES_PROVIDED += radio_nrf802154
+
+include $(RIOTBOARD)/common/nrf52/Makefile.features
+{% endhighlight %}
+
+The first line tells the build system, the CPU (or more accurate microcontoller)
+the board is using in a naming scheme that is understand by the build system. We
+basically tell it, "this board is using an nRF52840 microcontroller". Lines 4-9
+tell the build system which features are provided by the board, naming
+communication interfaces such as I²C, SPI, UART, and USB, but also that the
+radio of this board can speak the [IEEE 802.15.4] protocol—imagine something
+like low-power WiFi—via its nRF microcontroller.
+The last line declares to include further features provided by the nRF52 family
+of microcontrollers in general.
+
+Lastly, I tell the build system with the file
+[`boards/feather-nrf52840/Makefile.include`][Makefile.include], what else needs
+to be included for this board to be programmable, among them the most important
+thing: the programmer itself. In case of the Feather nRF52840 this is the Segger
+JLink:
+
+{% highlight Makefile %}
+# HACK: replicate dependency resolution in Makefile.dep, only works
+# if `USEMODULE` or `DEFAULT_MODULE` is set by the command line or in the
+# application Makefile.
+ifeq (,$(filter stdio_%,$(DISABLE_MODULE) $(USEMODULE)))
+  RIOT_TERMINAL ?= jlink
+endif
+
+include $(RIOTMAKE)/tools/serial.inc.mk
+
+include $(RIOTBOARD)/common/nrf52/Makefile.include
+{% endhighlight %}
+
+Now we finally leave the build system and tell the microcontroller where to put
```

Comment:
Now, ... 

"tell the microcontroller where to put all the peripherical interfaces to the outside world" -- needs rephrasing.